### ISSUE TITLE
Include a ConnectionString() extension method.

### ISSUE THREAD

**Issue by willwolfram18** [Particular/NServiceBus.Storage.MongoDB#16]
In [setting up the MongoDB persistence](https://docs.particular.net/persistence/mongodb/#usage), it is required that you build a `MongoClient` when you're connecting to a database that is not hosted on `mongodb://localhost:27017`. However, at least in my team's experience, we haven't needed a customize `MongoClient` for establishing connections when using [`NServieBus.MongoDB`](https://github.com/sbmako/NServiceBus.MongoDB). In fact, that project just performs a `new MongoClient(connectionString)` when [configuring a connection string](https://github.com/sbmako/NServiceBus.MongoDB/blob/master/src/NServiceBus.MongoDB/ConfigureMongoPersistence.cs#L162).

It would be wonderful if we could get a similar extension method, such as `ConnectionString(this PersistenceExtensions<MongoPersistence>, string)` that could perform the `new MongoClient()` work for us instead of needing to do this operation for each endpoint.

**Comment by timbussmann** [comment]
Thanks for the feedback @willwolfram18. I agree that the current configuration API feels noisy when you just want to pass a connection string.

**Comment by SzymonPobiega** [comment]
Hey @willwolfram18 

We looked at that issue in detail and we decided that we don't want to provide such a shortcut API. The main reason for us to refrain from adding it was the fact that making both method visible would be confusing for the users who would not know if they need to call both methods (and if so, if order matters) or only one of them. Another problem is how should the code behave is somebody calls both methods. Should the second one throw or last one win? We would like to avoid such complexities.

We used similar approach in SQL Server persistence where we require `Func<DbConnection>` instead of a connection string and it worked well for both simple and advanced cases.

Hope it makes sense.
